### PR TITLE
INFRA-5020 containers by status and longest running docker process

### PR DESCRIPTION
**Pull request by ahammond**
Adds `docker_container_status` and `docker_longest_running`

### CODE REVIEW

**Review by onorua — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -304,48 +379,69 @@ func zombieDockerProcessCount() (int, error) {
...
+			// The time the process started after system boot.
+			// ... snip ...
+			// Since Linux 2.6, the value is expressed
+			// in clock ticks (divide by sysconf(_SC_CLK_TCK)).
+			s := syscall.Sysinfo_t{}
+			err := syscall.Sysinfo(&s)
+			if err != nil {
+				log.Panic(err)
+			}
+			t := float64(s.Uptime) - float64(r.startTime) / clockTicksPerSec
```
Comment (onorua) [inline main.go]: shouldn't this be like:
```
(float64(s.Uptime) - float64(r.startTime)) / clockTicksPerSec
```

File: main.go [hunk 2]
```diff
@@ -106,8 +126,9 @@ func scrapeContainer(container types.Container, cli *client.Client, closer <-cha
 		return
 	}
 	name := inspect.Name
+	image_id := inspect.Image
```
Comment (white-hat) [inline main.go]: `imageID`?

File: main.go [hunk 3]
```diff
@@ -263,21 +342,29 @@ func newUnixProcess(pid int) (*UnixProcess, error) {
...
 	d, err := os.Open("/proc")
 	if err != nil {
-		return -1, err
+		return procResult{}, err
 	}
 	defer d.Close()
 
 	zombieCount := 0
+	cmdline := ""
+	smallestStartTime := int64(9223372036854775807)
```
Comment (white-hat) [inline main.go]: please use contants for magic numbers

File: main.go [hunk 4]
```diff
@@ -249,12 +295,45 @@ func (p *UnixProcess) Refresh() error {
...
+		&p.kstkesp,
+		&p.kstkeip,
+		&p.signal,
+		&p.blocked)
+
+	return err
+}
 
+func (p *UnixProcess) RefreshCmdline() error {
+	cmdlinePath := fmt.Sprintf("/proc/%d/cmdline", p.pid)
```
Comment (white-hat) [inline main.go:333]: it's it easier to check for existence of `cmdline` and return blank cmdline if it does not exist. Anyway, I having hard time to trace what happens now if this file does not exist, please double check that

**Review by white-hat — CHANGES_REQUESTED** [review]

File: main.go (hunk 4 above)
Comment (ahammond) [reply main.go:333]: Per http://man7.org/linux/man-pages/man5/proc.5.html the file always exists, but can be empty for zombies.

**Review by white-hat — APPROVED** [review]

**Pull request merged**